finetune: rename feed-forward tensors (w1/w2/w3)

danbev opened the pull request:
This commit renames the feed-forward tensors `w1`, `w2` and `w3` to `ffn_gate`, `ffn_down` and `ffn_up` respectively.

The motivation for this change is to make it easier to understand the purpose of the tensors. This also seems to be inline with the names used in the llama_layer struct in llama.cpp.

danbev committed:
finetune: rename feed-forward tensors (w1/w2/w3)

This commit renames the feed-forward tensors w1, w2 and w3 to ffn_gate,
ffn_down and ffn_up respectively.

The motivation for this change is to make it easier to understand the
purpose of the tensors. This also seems to be inline with the names
used in the llama_layer struct in llama.cpp.

Signed-off-by: Daniel Bevenius <daniel.bevenius@gmail.com>

xaedes reviewed (changes requested):
Looks good to me. 
For consistency, can you do the same for `examples/train-text-from-scratch.cpp` inside this PR?

danbev commented:
> For consistency, can you do the same for `examples/train-text-from-scratch.cpp` inside this PR?

Absolutely, I'll take a look at that examples as well :+1:

danbev committed:
train-text-from-scratch: rename ff tensors

This commit renames the feed-forward tensors w1, w2 and w3 to ffn_gate,
ffn_down and ffn_up respectively.

The motivation for this change is to make it easier to understand the
purpose of the tensors. This also seems to be inline with the names
used in the llama_layer struct in llama.cpp

Signed-off-by: Daniel Bevenius <daniel.bevenius@gmail.com>

danbev commented:
The [ci failure](https://github.com/ggerganov/llama.cpp/actions/runs/7471618039/job/20332210095?pr=4839#step:5:55) does not look related to this PR as far as I can tell. 
Would someone with the correct permissions be able to re-run the job in question?

ggerganov reviewed (commented):
Yeah, sometimes this job fails - restarted it

danbev commented:
@xaedes Would you be able to take a look at the changes to train-text-from-scratch.cpp? Thanks
